documentation: Update tutorial for writing help center documentation.

laurynmm opened the pull request:
Updates the tutorial for writing help center articles to encourage contributors to add to or enhance the existing help center docs before writing a new articles for new features. See [this CZO discussion](https://chat.zulip.org/#narrow/stream/19-documentation/topic/writing.20end.20user.20documentation) for more context.

Also, generally updates references to 'user documentation' to be 'help center documentation' or 'user-facing documentation'. The exception to this are any references to 'user documentation' in the 'Version history' section.

Additionally, updates some headers for clarity and consistency, and adds some links throughout the section 'Writing documentation'.

laurynmm commented:
Screenshots of updates to `user.md` (which potentially would be renamed?)

## 1. Updates to the start of the tutorial, removes 'Editing and Testing' header:
![Screenshot from 2021-12-21 14-29-20](https://user-images.githubusercontent.com/63245456/146945880-36ab5455-abda-4f52-a2ff-11c42b0dbc6e.png)

## 2. Addition of 'Getting started' section, emphasizes updates to current documentation:
![Screenshot from 2021-12-21 14-30-10](https://user-images.githubusercontent.com/63245456/146946008-4796dd2f-f679-49d7-8856-a09d0f45a2a6.png)

## 3. Creates 'Adding new article' section, with text from current tutorial
![Screenshot from 2021-12-21 14-30-41](https://user-images.githubusercontent.com/63245456/146946245-41a06d35-b394-4072-a4a1-856a4ed330cb.png)

## 4. Renames 'Features' to 'Documentation features' with some updates to list text
![Screenshot from 2021-12-21 14-31-04](https://user-images.githubusercontent.com/63245456/146946399-403b653f-b484-49c2-8e55-72597cdf0c54.png)

laurynmm commented:
Screenshot of sidebar with updates to section headers in `Writing documentation`.

I'm currently not sure if this is the best solution. I tried to match the naming convention of 'Documenting ...' but 'Documenting help center' doesn't work.

I went with ...
- Documenting Zulip's features
- Documenting an integration
- Documenting REST API endpoints

Another option would be to change the three sections to mirror the titles used in the 'User-facing documentation' section ...
- Help center documentation
- Integrations documentation
- API endpoint documentation

![Screenshot from 2021-12-21 14-28-38](https://user-images.githubusercontent.com/63245456/146947124-f9078656-d1f1-4129-9a0b-ee8c47da5afb.png)

timabbott commented on docs/documentation/overview.md:
```diff
@@ -2,27 +2,27 @@
   it or installing it), and written in Markdown.
 
-- Core website documentation: Complete webpages for complex topics,
+- **Core website documentation**: Complete webpages for complex topics,
```
Maybe we should rebrand this as "Product landing pages"?  I feel like "landing pages" is sorta a more typical name for what we mean by this sort of fancier page.

@alya thoughts?  Possibly you should just plan to do a pass on this paragraph after we merge this PR, since it's orthogonal to what's being changed here.

alya replied:
Do the landing pages even fall under "documentation"? It seems strange to categorize them as such, but I guess a contributor might update, e.g., the integrations page, which is why these pages are included here?

Anyway, yeah, let's bookmark that as a separate edit pass.

timabbott replied:
I think some of the more detailed ones like /features kinda fall into that category, but yeah, another option would be to delete this paragraph or replace it with just a brief mention that there's other user-facing content in landing pages and the like.

laurynmm replied:
I'm probably the exception, but I think I did end up interacting some with the core website documentation for my first PR with Zulip (which is still pending and perhaps wasn't the best choice for a first issue) as I updated some text on [this page of the Zulip site](https://zulip.com/apps/), so some sort of mention might be worth while. I don't think I really registered it as different from the other documentation pages at the time.

timabbott commented on docs/documentation/user.md:
```diff
@@ -1,36 +1,31 @@
-# User documentation
+# Documenting Zulip's features
```
Do we want to rebrand this to "Writing help center articles" or something similar to avoid having extra terminology for the same thing?

alya replied:
Sounds good to me.

laurynmm replied:
Yeah, that was my first idea, and we can go that route. I was trying to have this section's title be consistent with the other two documentation tutorials ("Documenting an integration" and "Documenting REST API endpoints"). Which all line up in the sidebar titles as you can see in the screenshot just above ...

laurynmm replied:
Here's a quick comparison of with "Writing help center articles" as the article title in the sidebar. I think it works, but perhaps we want another title for the sub-heading "Writing documentation"?
![Screenshot from 2022-01-06 15-59-21](https://user-images.githubusercontent.com/63245456/148402811-2eb577b6-1f45-44f5-82a1-9cb7305e6b3a.png)

timabbott commented on docs/documentation/user.md:
```diff
@@ -39,22 +34,63 @@ form `/help/foo`; with special cases for `/help/` going to `index.md` and
+  [features](#documentation-features) used to enchance the readability
+  of the documentation.
+
+- Could the feature you're documenting be added or merged into an
```
s/Could/Should/?  It's possible we should add a sentence of the form "If the new feature is a refinement on a feature that already has a dedicated page, it's likely more discoverable as a new section on that page than as an independent page, so that users can find the information by scrolling."

timabbott commented on docs/documentation/user.md:
```diff
@@ -65,17 +101,17 @@ Tips for adding a new article:
   to know?
 
+- Remember to explain the purpose of the feature and give context as well
+  as instructions.
```
Perhaps extend this sentence to "as well as instructions for how to use or enable it."?

timabbott commented on docs/documentation/user.md:
```diff
@@ -91,31 +127,40 @@ is needed. **Never identify or refer to a button by its color.**
+and used liberally.
 
-## Features
+## Documentation features
```
Would "Markdown features" be a more intuitive title?

timabbott commented:
I like these changes!  I would personally have tried to do this in a few commits, with the first being a straightforward "user documentation" => "help center" pass (so that the more interesting changes would stand out more when reviewing), but I definitely don't think you should redo things, since copywriting files like these are particularly bad on the merge conflicts front when trying to split commits.

I posted a few comments on naming / other details that came to mind.  There's probably more that we could do in terms of additions to help motivate or clarify our policies -- e.g. on the "Related articles" section, we might want to clarify policies for when to use inline links to related features in the context descriptions at the start of a section, and if we're doing that, whether we should also put that same link in a "Related articles" section at the bottom (My temptation would be that we should liberally include links and be happy to have a related article be linked both inline and at the bottom, since they're different ways of scanning for information).  But that's probably a decision we should only make and try to encode after a #documentation discussion.

But I don't think any of that should block merging these changes as a nice incremental improvement.

alya commented:
> (My temptation would be that we should liberally include links and be happy to have a related article be linked both inline and at the bottom, since they're different ways of scanning for information). But that's probably a decision we should only make and try to encode after a #documentation discussion.

We can discuss, but it should be noted that this has been the de facto policy, at least in the docs that I've been adding/editing.

alya commented on docs/documentation/user.md:
```diff
@@ -1,36 +1,31 @@
+- Public documentation of our feature set, for someone googling "can zulip do ..."
+
+- Quick responses to support questions; if someone emails a Zulip admin
   asking "how do I change my name", they can reply with a link to the doc.
```
I think it should be:

asking, "How do I change my name?" they can...

alya commented on docs/documentation/user.md:
```diff
@@ -39,22 +34,63 @@ form `/help/foo`; with special cases for `/help/` going to `index.md` and
+  link to as **Related articles** in your new feature documentation.
+
+Remember that real estate in the left sidebar is somewhat precious.
+Minor features should rarely get their own article, and instead be
```
and instead be -> and should instead be

alya commented:
I left some minor comments, but overall looks great!

laurynmm commented:
> e.g. on the "Related articles" section, we might want to clarify policies for when to use inline links to related features in the context descriptions at the start of a section, and if we're doing that, whether we should also put that same link in a "Related articles" section at the bottom

Just had a light bulb moment that what's missing is a section dedicated to "Updating an existing article", which should come before the section on "Adding a new article". It can include the clarification about linking inline contextually as well as including a "Related articles". 

FWI I feel like the updated article for "Configure default view" when my implementation of the escape key navigation was added is a great example of the extending existing articles for feature additions. But I don't know if we want to avoid using documentation pages as examples because they might be changed in the future.

laurynmm commented:
@timabbott and @alya - I've updated the original commit with your edits and comments above.

I also added a section to the tutorial content: 'Updating an existing article'. It's in a separate commit so that you can review this new content addition easily. Once I have your feedback on whether that is a useful addition, I can squash the commits (or remove the second one) and get the PR ready for final review.

One additional question I have is updating the name of the file `docs/documentation/user.md`. Do we want to change that to `helpcenter.md`? Or keep it as is?

Here's a screenshot of the major content add in the second commit with the sidebar extended so you can see the header changes as well:

![Screenshot from 2022-01-11 17-05-27](https://user-images.githubusercontent.com/63245456/148979258-3393d25a-1b93-4ae5-9139-5c11eeea5f3c.png)

alya commented on docs/documentation/user.md:
```diff
@@ -83,10 +86,40 @@ can always ask in
+
+Creating robust and informative help center articles with good links
+will allow users to navigate the help center much more effectively
+and efficiently.
```
I would reword this to, "navigate the help center more effectively" for brevity.

alya commented:
I left a couple of small comments, but overall looks good!

I haven't re-reviewed the rest of the PR. @timabbott let me know if I need to.

timabbott commented:
> One additional question I have is updating the name of the file docs/documentation/user.md. Do we want to change that to helpcenter.md? Or keep it as is?

I think probably `helpcenter.md` would be nicer, yeah.  I'd do it in an extra commit -- it'll help remind me to add a redirect in ReadTheDocs when we merge this.

laurynmm commented:
Marking this as ready to review now. I have added a separate commit for the name change of `user.md` to `helpcenter.md`.

Follow-up task from the comments above is to revisit the mention of `core website documentation` at the start of `docs/documentation/overview.md`.

timabbott commented:
This is great, merged, thanks @laurynmm!
